### ISSUE TITLE
Document the required function registration and parameters for various algorithm types

### ISSUE THREAD

**Issue by nhorman** [openssl/project#1002]
Add-on option: Enhance shells with documentation (and maybe template
code) as to which param-implementing getter/setter functions and dispatch table entries
need to be present to ensure proper operation (i.e., which of the many OSSL_PARAM
options are required for operations and which ones are merely just that,
options)

Documentation for getter/setter OSSL_PARAMs should go to per-alg man pages
documentation for dispatch functions should go to man7/provider-<alg> doc files

**Comment by baentsch** [comment]
fwiw, I don't think this is strictly a prerequisite for ML-KEM merging given that there are several provider implementations now getting this right.

**Comment by levitte** [comment]
As for `OSSL_PARAM`, this was originally designed to be all about optional values...  unfortunately, it turned out that some of them became mandatory, such as the cipher block size, which libcrypto code needs to be able to function properly.

**Comment by baentsch** [comment]
> some of them became mandatory,

Yup -- but only very few. And in my opinion, they are already documented as such, e.g., in https://docs.openssl.org/master/man7/provider-keymgmt/#common-information-parameters: Wording is an IMO clear "should be" -- might be strengthened to a "must be", e.g., 

> "max-size" (OSSL_PKEY_PARAM_MAX_SIZE) <integer>
> 
> The value should be the maximum size that a caller should allocate to safely store ....

Given that the new providers are basically done without this, worthwhile closing this issue? Or should this be retained/moved to "nice to improve as time permits" issues enhancing the wording? Or really add a separate page documenting which params _must_ be present? If the latter, suggestions welcome where that should be @levitte -- I'd then volunteer doing a PR.
